Rebus.AzureQueues/Config/AzureStorageQueuesConfigurationExtensions.cs

```diff
@@ -71,17 +90,25 @@ public static void UseAzureStorageQueues(this StandardConfigurer<ITransport> con
             Register(configurer, inputQueueAddress, storageAccount, options);
         }
 
-        static void Register(StandardConfigurer<ITransport> configurer, string inputQueueAddress, CloudStorageAccount storageAccount, AzureStorageQueuesTransportOptions optionsOrNull)
+
+        static void Register(StandardConfigurer<ITransport> configurer, string inputQueueAddress,
+            CloudStorageAccount cloudStorageAccount, AzureStorageQueuesTransportOptions optionsOrNull)
+        {
+            var queueFactory = new CloudQueueClientQueueFactory(cloudStorageAccount.CreateCloudQueueClient());
+            Register(configurer, inputQueueAddress, queueFactory, optionsOrNull);
+        }
+
+        static void Register(StandardConfigurer<ITransport> configurer, string inputQueueAddress, ICloudQueueFactory queueFactory, AzureStorageQueuesTransportOptions optionsOrNull)
         {
             if (configurer == null) throw new ArgumentNullException(nameof(configurer));
-            if (storageAccount == null) throw new ArgumentNullException(nameof(storageAccount));
+            if (queueFactory == null) throw new ArgumentNullException(nameof(queueFactory));
 
             var options = optionsOrNull ?? new AzureStorageQueuesTransportOptions();
 
             configurer.Register(c =>
             {
                 var rebusLoggerFactory = c.Get<IRebusLoggerFactory>();
-                return new AzureStorageQueuesTransport(storageAccount, inputQueueAddress, rebusLoggerFactory, options);
+                return new AzureStorageQueuesTransport(queueFactory, inputQueueAddress, rebusLoggerFactory, options);
```

Comment:
If the cloud queue client factory was in the container, then it would probably look like this:
```csharp

configurer.Register(c => {
	if (!configurer.Has<ICloudQueueFactory>())
	{
		configurer.Register<ICloudQueueFactory>(_ => new CloudQueueClientQueueFactory(...));
	}

	var rebusLoggerFactory = c.Get<IRebusLoggerFactory>();
	var queueFactory = c.Get<ICloudQueueFactory>();
	
	return new AzureStorageQueuesTransport(queueFactory, inputQueueAddress, rebusLoggerFactory, options);
})

```

Reply:
Is this relevant, considering the other conversation? `ICloudQueueFactory` would be passed as an argument, so no reason to resolve it from the container. We could still register it, but is there a purpose in doing so?

Reply:
no purpose in registering it if it is passed as an argument....